.circleci/config.yml

```diff
@@ -181,6 +181,16 @@ commands:
       - store_artifacts:
           path: << parameters.log_dir >>/<< parameters.env >>/test-results
           destination: results
+      - run:
+          working_directory: integration-tests
+          name: Test results Slack notification on Dev branch
+          command: |
+            if [ "$CIRCLE_NODE_INDEX" -eq 0 ] && [ "$CIRCLE_BRANCH" = "dev" ]; then
```

Comment:
Run this additional step only if `CIRCLE_NODE_INDEX` is 0 and on Dev branch because we can't have all nodes excute the same script.
Chose `CIRCLE_NODE_INDEX=0` because this node always exists: works for parallelism >= 1.

Reply:
<img width="1181" alt="Screen Shot 2022-08-12 at 11 52 17 AM" src="https://user-images.githubusercontent.com/35533885/184394627-796c2f35-23c5-4ec0-9766-5f9dbccdcafe.png">

Reply:
Would putting this in a `when` step instead of checking a condition in the script make it clearer what's happening in the CircleCI UI?

https://circleci.com/docs/configuration-reference#the-when-step-requires-version-21

Reply:
With `when:`:
```
      - when:
          and:
            - equal: [ $CIRCLE_NODE_INDEX, 0 ]
            - equal: [ $CIRCLE_BRANCH, "dev" ]
              steps:
                - run:
                    working_directory: integration-tests
                    name: Test results Slack notification on Dev branch
                    command: |
                      bash ../.circleci/wait-for-job-finish.sh
                      node --require ../.pnp.cjs ./slack/notify-circleci-test-results.js
          when: always
```

Reply:
Couldn't get `when` to work in step. I found an [explanation](https://discuss.circleci.com/t/using-environment-variables-inside-a-conditional/39152/4) from CircleCI help.

> Because _**when**_ statements are evaluated when the configuration is compiled prior to running. Environment variables are expanded at runtime.